esp/services/ws_workunits/ws_wudetails.hpp

```diff
@@ -0,0 +1,126 @@
+        unsigned __int64 getMaxValue()   { return maxValue; }
+        bool hasExactValue()             { return bExactValue; }
+        bool hasMinValue()               { return bMinValue; }
+        bool hasMaxValue()               { return bMaxValue; }
+    };
+
+    static int funcCompareAttribFilter(IInterface* const * l, IInterface* const * r)
+    {
+        CAttributeFilter * ll = (dynamic_cast<CAttributeFilter *>(*l));
+        CAttributeFilter * rr = (dynamic_cast<CAttributeFilter *>(*r));
+
+        if (ll->getStatisticKind()>rr->getStatisticKind()) return +1;
+        if (ll->getStatisticKind()<rr->getStatisticKind()) return -1;
+        return 0;
+    }
+    static int funcCompareUnsignedShort(const unsigned short * l, const unsigned short * r)
+    {
+        if (*l>*r) return +1;
+        if (*l<*r) return -1;
+        return 0;
+    }
+    static int funcCompareString(char const * const *l, char const * const *r)
+    {
+      return strcmp(*l, *r);
+    }
+    StatisticsFilter * initStatisticsFilter(IConstWUScopeFilter & filter,
+                                            IConstWUAttributeToReturn & attribsToReturn);
+
+    void buildAttribFilter(IArrayOf<CAttributeFilter> & processedAttribFilter,
+                           IArrayOf<IConstWUAttributeFilter> & reqAttribFilter);
+
+    void buildAttribListToReturn(UnsignedShortArray & skReturnAttribList,
+                                 StringArray & returnAttriblist);
+
+    void buildScopeTypeList(UnsignedShortArray & processedScopeTypeList,
+                            StringArray & arrayScopeTypeList);
+
+    void processWUStatsIntoResponse(IArrayOf<IEspWUScope> & respScopes,
+                                    IConstWUStatisticIterator & statsIter,
+                                    UnsignedShortArray & nestedScopeTypes,
+                                    UnsignedShortArray & returnAttribList,
+                                    StringArray & ids,
+                                    IArrayOf<CAttributeFilter> & arrayAttributeFilter,
+                                    unsigned __int64 minVersion,
+                                    unsigned __int64 & maxTimestamp);
+
+    IEspWUScope *buildRespWUScope(SCMStringBuffer & scope,
+                                  SCMStringBuffer & scopeType,
+                                  SCMStringBuffer & id,
+                                  IArrayOf<IEspWUResponseAttribute> & attribs);
+
+    IEspWUResponseAttribute *copyAttribs(IConstWUStatistic & cur,
+                                         IConstWUAttributeOptions & attribOptions);
+
+    SCMStringBuffer & getScopeIdFromScopeName(SCMStringBuffer & scopeId,
+                                              SCMStringBuffer & scopeName);
+    inline unsigned getScopeDepth(SCMStringBuffer & sScope)
+    {
+        unsigned depth = 1;
+
+        for (const char * p = sScope.str(); *p; ++p)
+            if (*p==':') ++depth;
+        return depth;
+    }
+
+public:
```

Comment:
style:  Others may disagree with me on this point!  
In general it makes much more sense for the public portion of a class to come first.  Otherwise anyone who wants to use this class needs to scan through the implementation details before they can find the public interface.